rules/framework/vfs_overlay.bzl

```
@@ -62,15 +65,16 @@ def _build_subtrees(paths, vfs_prefix):
         # Loop the _framework_ path and add each dir to the current tree.
         # Assume the last bit is a file then add it as a file
         idx = 0
+
+        parts_len = len(parts)
         for part in parts:
-            if idx == len(parts) - 1:
-                ext_c = _get_external_contents(vfs_prefix, path_info.path)
+            if idx == parts_len - 1:
                 curr_subdirs.dict[part] = -1
-                curr_subdirs.json["contents"].append({"name": part, "type": "file", "external-contents": ext_c})
+                curr_subdirs.json["contents"].append({"name": part, "type": "file", "external-contents": vfs_prefix + path_info.path})
                 break
 
             # Lookup a value for the current subdirs, otherwise append
-            next_subdirs = curr_subdirs.dict.get(part, None)
+            next_subdirs = curr_subdirs.dict[part] if part in curr_subdirs.dict else None
```

My guess is that this operates the same as before but is just a bit harder to read so we can just leave it as is:

```suggestion
            next_subdirs = curr_subdirs.dict.get(part, None)
```

Starlark Go implementation:

https://github.com/google/starlark-go/blob/4b1e35fe22541876eb7aa2d666416d865d905028/starlark/library.go#L1193-L1207

---

yep, in Luis's comment, but I'm also with keeping it since there's evidence that it performs better